src/Action/InitElasticSearch.php

```diff
@@ -106,38 +115,40 @@ public function __invoke($params)
             $config['hosts'][0]['pass'] = array_shift($params);
         }
 
-        $taoVersion = $this->getServiceLocator()->get(\common_ext_ExtensionsManager::SERVICE_ID)->getInstalledVersion('tao');
-
-        if (version_compare($taoVersion, '7.8.0') < 0) {
-            return new Report(Report::TYPE_ERROR, 'Requires Tao 7.8.0 or higher');
-        }
-
         $oldSearchService = $this->getServiceLocator()->get(Search::SERVICE_ID);
         $oldSettings = $oldSearchService->getOptions();
 
         if (isset($oldSettings['settings'])) {
             $config['settings'] = $oldSettings['settings'];
         }
 
-        $isMap = false;
-        if (isset($oldSettings['isMap'])) {
-            $isMap = $oldSettings['isMap'];
-        }
-        $config['isMap'] = $isMap;
+        $config[GenerisSearch::class] = new GenerisSearch();
 
-        $search = new ElasticSearch($config);
-        $search->flush();
+        try {
+            $search = new ElasticSearch($config);
 
-        if ($isMap) {
-            $search->settingUpIndexes();
+            $search->createIndexes();
+            $this->getServiceManager()->register(Search::SERVICE_ID, $search);
+
+            $report->add(new Report(Report::TYPE_SUCCESS, __('Switched search service implementation to ElasticSearch')));
+
+            $this->getServiceManager()->register(IndexUpdaterInterface::SERVICE_ID, new IndexUpdater($config['hosts']));
+        } catch (BadRequest400Exception $e) {
+            $report->add(new Report(Report::TYPE_ERROR, 'Unable to create index: ' . $e->getMessage()));
+        } catch (\Exception $e) {
+            $report->add(new Report(Report::TYPE_ERROR, 'ElasticSearch server could not be found'));
```

Comment:
Same here - you can use static methods to create `Report` with the necessary type